src/runtime/vulkan_extensions.h

```diff
@@ -0,0 +1,298 @@
+#ifndef HALIDE_RUNTIME_VULKAN_EXTENSIONS_H
+#define HALIDE_RUNTIME_VULKAN_EXTENSIONS_H
+
+#include "vulkan_internal.h"
+
+// --------------------------------------------------------------------------
+
+namespace Halide {
+namespace Runtime {
+namespace Internal {
+namespace Vulkan {
+
+// --------------------------------------------------------------------------
+
+WEAK char layer_names[1024];
+WEAK ScopedSpinLock::AtomicFlag layer_names_lock = 0;
+WEAK bool layer_names_initialized = false;
+
+WEAK char extension_names[1024];
+WEAK ScopedSpinLock::AtomicFlag extension_names_lock = 0;
+WEAK bool extension_names_initialized = false;
+
+WEAK char device_type[256];
+WEAK ScopedSpinLock::AtomicFlag device_type_lock = 0;
+WEAK bool device_type_initialized = false;
+
+WEAK char build_options[1024];
+WEAK ScopedSpinLock::AtomicFlag build_options_lock = 0;
+WEAK bool build_options_initialized = false;
+
+// --------------------------------------------------------------------------
+namespace {
+
+void vk_set_layer_names_internal(const char *n) {
+    if (n) {
+        size_t buffer_size = sizeof(layer_names) / sizeof(layer_names[0]);
+        strncpy(layer_names, n, buffer_size);
```

Comment:
...I know that our opencl runtime already uses strncpy, but we should probably avoid adding new uses of it, since it requires the "add a null terminator just in case" nonsense that is easy to forget.